test/unit/ContainerSetPathCapableTraitTest.php

```diff
@@ -0,0 +1,197 @@
+    {
+        $mock = $this->getMockBuilder('Psr\Container\ContainerInterface')
+            ->setMethods($methods)
+            ->getMock();
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $message The exception message.
+     *
+     * @return Exception The new exception.
+     */
+    public function createException($message = '')
+    {
+        $mock = $this->getMockBuilder('Exception')
+            ->setConstructorArgs([$message])
+            ->getMock();
+
+        return $mock;
+    }
+
+    /**
+     * Creates a new Not Found exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $message The exception message.
+     *
+     * @return MockObject|RootException|InvalidArgumentException The new exception.
+     */
+    public function createInvalidArgumentException($message = '')
+    {
+        $mock = $this->getMockBuilder('InvalidArgumentException')
+            ->setConstructorArgs([$message])
+            ->getMockForAbstractClass();
+
+        return $mock;
+    }
+
+    /**
+     * Merges the values of two arrays.
+     *
+     * The resulting product will be a numeric array where the values of both inputs are present, without duplicates.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $destination The base array.
+     * @param array $source      The array with more keys.
+     *
+     * @return array The array which contains unique values
+     */
+    public function mergeValues($destination, $source)
+    {
+        return array_keys(array_merge(array_flip($destination), array_flip($source)));
+    }
+
+    /**
+     * Tests whether a valid instance of the test subject can be created.
+     *
+     * @since [*next-version*]
+     */
+    public function testCanBeCreated()
+    {
+        $subject = $this->createInstance();
+
+        $this->assertInternalType(
+            'object',
+            $subject,
+            'A valid instance of the test subject could not be created.'
+        );
+    }
+
+    /**
+     * Tests that `_containerSetPath()` works as expected.
+     *
+     * @since [*next-version*]
+     */
+    public function testContainerSetPathSuccessfully()
+    {
+        $key1 = uniqid('key1');
+        $key2 = uniqid('key2');
+        $val = uniqid('val');
+        $newVal = uniqid('new-val');
+        $path = [$key1, $key1];
+        $data = [
+            $key1 => [
+                $key2 => $val,
+            ],
+        ];
+        $container = (object) $data;
+        $subject = $this->createInstance(['_normalizeArray', '_containerSet']);
+
+        $subject->expects($this->exactly(count($path)))
+            ->method('_normalizeArray')
+            ->will($this->returnArgument(0));
+
+        $subject->expects($this->exactly(1))
+            ->method('_containerSet');
+
+        $reflection = new ReflectionMethod($subject, '_containerSetPath');
```

Comment:
Here and in other test, use `Xpmock` reflection method.

Reply:
Can't do that, because `Xpmock` will not allow passing arguments by reference, which is required by the SUT.